src/cloud/ops.rs

```diff
@@ -124,80 +121,24 @@ pub async fn create_cloud_instance(
     Ok(())
 }
 
-fn ask_name() -> anyhow::Result<String> {
-    let instances = credentials::all_instance_names()?;
-    loop {
-        let name = question::String::new(
-            "Specify a name for the new instance"
-        ).ask()?;
-        if !is_valid_instance_name(&name) {
-            echo!(err_marker(),
-                "Instance name must be a valid identifier, \
-                 (regex: ^[a-zA-Z_][a-zA-Z_0-9]*$)");
-            continue;
-        }
-        if instances.contains(&name) {
-            echo!(err_marker(),
-                "Instance", name.emphasize(), "already exists.");
-            continue;
-        }
-        return Ok(name);
-    }
-}
-
-pub fn split_cloud_instance_name(name: &str) -> anyhow::Result<(String, String)> {
-    let mut splitter = name.splitn(2, '/');
-    match splitter.next() {
-        None => unreachable!(),
-        Some("") => anyhow::bail!("empty instance name"),
-        Some(org) => match splitter.next() {
-            None => anyhow::bail!("cloud instance must be in the form ORG/INST"),
-            Some("") => anyhow::bail!("invalid instance name: missing instance"),
-            Some(inst) => Ok((String::from(org), String::from(inst))),
-        },
-    }
-}
-
-pub async fn create(
-    cmd: &crate::portable::options::Create,
-    opts: &crate::options::Options,
-) -> anyhow::Result<()> {
-    let client = CloudClient::new(&opts.cloud_options)?;
-    client.ensure_authenticated(false)?;
-
-    let name = if let Some(name) = &cmd.name {
-        name.to_owned()
-    } else if cmd.non_interactive {
-        echo!(err_marker(), "Instance name is required \
-                             in non-interactive mode");
-        return Err(ExitCode::new(2).into());
-    } else {
-        ask_name()?
-    };
-
-    let (org, inst_name) = split_cloud_instance_name(&name)?;
-    let instance = CloudInstanceCreate {
-        name: inst_name.clone(),
-        org,
-        // version: Some(format!("{}", version.display())),
-        // default_database: Some(cmd.default_database.clone()),
-        // default_user: Some(cmd.default_user.clone()),
-    };
-    create_cloud_instance(&client, &instance).await?;
-    print::echo!(
-        "EdgeDB Cloud instance",
-        name.emphasize(),
-        "is up and running."
+pub async fn prompt_cloud_login(client: &mut CloudClient) -> anyhow::Result<()> {
+    let mut q = question::Confirm::new(
+        "You're not authenticated to the EdgeDB Cloud yet, login now?",
     );
-    print::echo!("To connect to the instance run:");
-    print::echo!("  edgedb -I", name);
-    Ok(())
+    if q.default(true).ask()? {
+        crate::cloud::auth::do_login(&client).await?;
+        client.reinit()?;
```

Comment:
Yeah it was a new client instance, but I would create new client from `ask_name()` and use it from the outside caller, just doesn't feel clean to pass the client around, rather than a `&mut client` with `reinit()`.

Reply:
Well, actually you may do `*client = Client::new()`. Replace is only needed when you need old object around :)